Source/Hypermedia.Client/Hypermedia/SelfHypermediaLink.cs

```diff
@@ -0,0 +1,19 @@
+using System;
+using System.Threading.Tasks;
+
+namespace Bluehands.Hypermedia.Client.Hypermedia
+{
+    public class SelfHypermediaLink<T> : MandatoryHypermediaLink<T> where T : HypermediaClientObject
+    {
+        public new async Task<T> ResolveAsync(bool ignoreCache = true)
```

Comment:
using new here is not the best of styles, but considering we are planning to move ResolveAsync from class method to extension method, this is resolved then by offering this extension on the most specified type SelfHypermediaLink

Reply:
True, "new" smells. But it is only intermediate. It should not stay in the final version.